mypy/types.py

```diff
@@ -1162,23 +1162,45 @@ class TypeType(Type):
     # a generic class instance, a union, Any, a type variable...
     item = None  # type: Type
 
+    # this class should not be created directly
+    # use make method which may return either TypeType or UnionType
+    # this is to ensure Type[Union[A, B]] is always represented as Union[Type[A], Type[B]]
     def __init__(self, item: Type, *, line: int = -1, column: int = -1) -> None:
```

Comment:
What I'm struggling with is, how to tell at the call site (where I create `TypeType`) whether it should use a regular constructor or a `make_normalized`. In most cases, I only know the argument is of type `Type`, not which particular subtype it is. So I end up having to call `make_normalized` everywhere. And that's dangerous because in my `make_normalized` I have to crash mypy if an unexpected type appears (see my last commit).

I still like the approach without `__new__`, I just can't figure out how to do it safely.